pkg/synopsysctl/cmd_destroy.go

```diff
@@ -74,51 +79,59 @@ var destroyCmd = &cobra.Command{
-	} else {
-		data := cm.Data["config.json"]
-		var cmData map[string]interface{}
-		err = json.Unmarshal([]byte(data), &cmData)
-		if err != nil {
-			log.Errorf("unable to unmarshal config map data due to %+v", err)
-		}
-		if crdNames, ok := cmData["CrdNames"]; ok {
-			crds = strings.Split(crdNames.(string), ",")
-		}
-	}
-
 	// delete namespace
-	checkErr := util.CheckResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
-	if err != nil && isForceDestroy {
+	isNamespaceExist, err := util.CheckResourceNamespace(kubeClient, namespace, "", true)
+	if isNamespaceExist && err != nil && isForceDestroy {
 		log.Warnf("%s. namespace cannot be deleted", err.Error())
-	} else if err != nil && !isForceDestroy {
+	} else if isNamespaceExist && err != nil && !isForceDestroy {
 		return fmt.Errorf("%s. It is not recommended to destroy the Synopsys Operator so these resources can continue to be managed. If you are sure you want to delete the Synopsys Operator anyway then you can use the 'force' option which will keep all the instances and delete only the Synopsys Operator", err.Error())
+	} else if err != nil {
+		log.Error(err)
 	}
 
-	log.Infof("deleting the Synopsys Operator in namespace '%s'", namespace)
 	// delete Synopsys Operator instance
-	commonConfig := crdupdater.NewCRUDComponents(restconfig, kubeClient, false, false, namespace, &api.ComponentList{}, "app=synopsys-operator", false)
-	_, crudErrors := commonConfig.CRUDComponents()
-	if len(crudErrors) > 0 {
-		log.Errorf("unable to delete the Synopsys Operator in namespace '%s' due to %+v", namespace, err)
-	}
+	if isNamespaceExist {
+		// get the synopsys operator configmap to get the crd names and cluster scope
+		crds := []string{}
+		cm, cmErr := util.GetConfigMap(kubeClient, namespace, "synopsys-operator")
+		if cmErr != nil {
+			log.Errorf("error getting the config map in namespace '%s' due to %+v", namespace, cmErr)
+		} else {
+			data := cm.Data["config.json"]
+			var cmData map[string]interface{}
+			cmErr = json.Unmarshal([]byte(data), &cmData)
+			if cmErr != nil {
+				log.Errorf("unable to unmarshal config map data due to %+v", cmErr)
+			}
+			if crdNames, ok := cmData["CrdNames"]; ok {
+				crds = strings.Split(crdNames.(string), ",")
+			}
+		}
 
-	if checkErr == nil {
-		err = util.DeleteNamespace(kubeClient, namespace)
-		if err != nil {
-			log.Errorf("unable to delete the Synopsys Operator namespace due to %+v", err)
+		log.Infof("deleting the Synopsys Operator resources in namespace '%s'", namespace)
+		if err == nil {
+			err = util.DeleteNamespace(kubeClient, namespace)
+			if err != nil {
+				log.Errorf("unable to delete the Synopsys Operator namespace due to %+v", err)
+			}
+		}
+
+		commonConfig := crdupdater.NewCRUDComponents(restconfig, kubeClient, false, false, namespace, "", &api.ComponentList{}, "app=synopsys-operator", false)
```

Comment:
Deleting the NS will delete all resources in the NS, so this should only be done in NS mode I think?  Should be in an else block I think.

Reply:
done